bin/install_cocoapods

```diff
@@ -13,13 +13,37 @@ LOCAL_CACHE_KEY="$BUILDKITE_PIPELINE_SLUG-local-pod-cache-$PODFILE_HASH"
 restore_cache "$LOCAL_CACHE_KEY"
 
 # If the `pod check` plugin is installed, use it to determine whether or not to install Pods at all
-# If it's not installed (or if it fails), we'll try to install Pods. 
+# If it's not installed (or if it fails), we'll try to install Pods.
 # If that fails, it may be due to an out-of-date repo. We can use `--repo-update` to try to resolve this automatically.
 if bundle exec pod plugins installed | grep -q check; then
 	bundle exec pod check || bundle exec pod install || bundle exec pod install --repo-update --verbose
 else
 	bundle exec pod install || bundle exec pod install --repo-update --verbose
 fi
 
+# Check that `Podfile.lock` was unchanged by `pod install`. If it was, it means
+# the lockfile might have been inadvertently changed.
+
+# `shasum` is available only on macOS
+if command -v shasum &> /dev/null; then
+  sha_command='shasum -a 256'
+else
+  sha_command='sha256sum'
+fi
+
+function lockfile_error () {
+  message="\`Podfile.lock\` was changed by \`bundle exec pod install\`. Please run \`bundle exec pod install\` locally with a clean cache and commit the result."
+  echo "$message"
+  buildkite-agent annotate "$message" --style 'error' --context 'ctx-error'
+}
+trap lockfile_error ERR
```

Comment:
Nice use of `trap` here! 👍 👌

Reply:
Thanks 😄 I copy pasted it [from the CircleCI orb](https://github.com/wordpress-mobile/circleci-orbs/blame/25eee7663ad3d9fc908c7602ae3262379ea33c61/src/ios/orb.yml#L501-L504)... Kudos to @jtreanor .